### PR TITLE
[python-nextgen] Add py.typed

### PR DESCRIPTION
**Pull request by wing328**
Add py.typed

To close https://github.com/OpenAPITools/openapi-generator/issues/15097

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date.sh
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

FYI @taxpon (2017/07) @frol (2017/07) @mbohlool (2017/07) @cbornet (2017/09) @kenjones-cisco (2017/11) @tomplus (2018/10) @arun-nalla (2019/11) @spacether (2019/11) @krjakbrjak (2023/02)

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/python-nextgen/setup.mustache [hunk 1]
```diff
@@ -49,6 +49,7 @@ setup(
     long_description="""\
     {{appDescription}}  # noqa: E501
     """
```
Comment (robertschweizer) [inline modules/openapi-generator/src/main/resources/python-nextgen/setup.mustache:51]: I think this is missing a comma:

```suggestion
    """,
```

**Comment by wing328** [comment]
@robertschweizer thanks for the review. Merged your fix and updated the samples.

**Comment by robertschweizer** [comment]
Thanks for adding `py.typed`! 

I don't know of any option in mypy to ignore a specific third-party package (maybe it's better for other type checkers). For some users, this change could be "breaking" if their type checker starts failing.

TBH I haven't really run mypy on our codebase using the new client yet. At first glance there are a few glitches still in the type annotations. For example, we can pass dictionaries to request functions instead of model objects, but the type annotations of the request functions only allow model objects. Should I open an issue for this?

**Review by tomplus — APPROVED** [review]

**Review by Jay-Madden — APPROVED** [review]
Running into this need right now attempting to get mypy working with a generated python client

**Comment by wing328** [comment]
closed via https://github.com/OpenAPITools/openapi-generator/pull/15804

**Pull request closed without merging**